### PR TITLE
add KGW algrithm implementation demo based on latest vllm version (vllm v1)

### PR DESCRIPTION
**Pull request by yiming2002**
Existing vLLM demo(MarkvLLM_demo.py) are incompatible with the recent vLLM v1(vllm package ver: 0.16.0) architectural update (v0 is now legacy).
I added a new KGW implementation demo optimized for vLLM v1, with a vLLM_v1 style KGW logits processor.
This demo may have throughput improvement over the Hugging Face Transformers baseline and even the old vLLM demo.
Can be tested using: python MarkvLLM_v1_demo.py.
Thank you for maintaining this project!

### CODE REVIEW

File: watermark/kgw/kgw_logits_processor_for_vllm.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            req = self.req_info.get(i)
+            if req is None:
+                mask[i] = 0
+                output_ids_matrix[i] = torch.zeros(window_size, dtype=torch.long, device=logits.device)
+                continue
+            else:
+                window_ids = req["prompt_ids"] + req["output_ids_in_window"]
+                if(len(window_ids) < window_size):
+                    return logits
+                window_ids = window_ids[-window_size:]
```
[the remark this answers is not in the document]
Comment (yiming2002) [reply watermark/kgw/kgw_logits_processor_for_vllm.py]: This does not need to be handled per request, because the upstream batching logic guarantees that all requests in the same batch use the same KGW parameters, so reading window_size, gamma, and delta once at the batch level is sufficient.

File: watermark/kgw/kgw_logits_processor_for_vllm.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    def apply(self, logits: torch.Tensor) -> torch.Tensor:
+        if not self.req_info:
+            
+            return logits
+
+        batch_size = logits.shape[0]
+        window_size = next(iter(self.req_info.values()))["prefix_length"]
+        gamma = next(iter(self.req_info.values()))["gamma"]
+        delta = next(iter(self.req_info.values()))["delta"]
+
```
[the remark this answers is not in the document]
Comment (yiming2002) [reply watermark/kgw/kgw_logits_processor_for_vllm.py]: This does not need to be handled per request, because the upstream batching logic guarantees that all requests in the same batch use the same KGW parameters, so reading window_size, gamma, and delta once at the batch level is sufficient.
